internal/rule/condition.go

```diff
@@ -3,82 +3,112 @@ package rule
 import (
 	"fmt"
 	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/filter"
+	"math"
 	"time"
 )
 
+// RetryNever indicates that an escalation condition should never be retried once it has been evaluated.
+const RetryNever = time.Duration(math.MaxInt64)
+
 type EscalationFilter struct {
 	IncidentAge      time.Duration
 	IncidentSeverity event.Severity
 }
 
-func (c *EscalationFilter) EvalEqual(key string, value string) (bool, error) {
+// ExtractRetryAfter extracts a time.Duration from the given filter conditions relative to the actual incident age.
+//
+// The retry after duration is extracted from the specified escalation condition that evaluate the "incident_age"
+// filter column. This function returns rule.RetryNever when all the incident age filter values are less than
+// the actual incident age or none of the filter conditions evaluates the "incident_age" column. Otherwise,
+// a time.Duration < rule.RetryNever and > the current incident age at which this escalation condition should
+// be re-evaluated is returned.
+func (e *EscalationFilter) ExtractRetryAfter(escalationCond filter.Filter) time.Duration {
```

Comment:
My suggestion for the overall function documentation (including a renaming of the function itself) would be the following:

```suggestion
// ReevaluateAfter returns the duration after which escalationCond should be reevaluated the 
// next time on the incident represented by e.
//
// escalationCond must correspond to an escalation that did not trigger on the incident
// represented by e before. If nothing in the incident changes apart from time passing by,
// the escalation is guaranteed to not trigger within the returned duration. After that
// duration, the escalation should be reevaluated and it may or may not trigger. If anything
// else changes, for example due to an external event, the escalation must be reevaluated as
// well.
func (e *EscalationFilter) ReevaluateAfter(escalationCond filter.Filter) time.Duration {
```

Parts of the current comment could be moved to inline comments within the implementation to explain it if you deem that necessary. But my impression is that it just rephrases the code in human language without giving much extra explanation.